fission-web-server/library/Fission/Web/Server/PowerDNS.hs

```diff
@@ -0,0 +1,14 @@
+module Fission.Web.Server.PowerDNS
+  ( module Fission.Web.Server.PowerDNS.Class
+  , getValuesFromRecords) where
+
+import           Fission.Prelude
+import           Fission.Web.Server.PowerDNS.Class
+
+import           Data.List                         (head)
+import           PowerDNS.Client                   as PDNS
+
+getValuesFromRecords :: [SearchResult] -> Maybe (NonEmpty Text)
+getValuesFromRecords rrs = case rrs of
+  []     -> Nothing
+  result -> Just $ pure (PDNS.sr_content $ head result)
```

Comment:
I see the Schemer in you coming out 😁

`head` is generally avoided, because it's a partial function; if you run `head []`, it'll blow up. `NonEmpty.head :: NonEmpty a -> a` on the other hand is totally safe, because it guarantees that there's at least one value (look at the constructor to see why)

In this case, it's more common to use pattern matching:

```haskell
case rrs of
  [] -> Nothing
  (h : _) -> Just . pure $ PDNS.sr_content h
```

There's also a helper function for this: [`nonEmpty :: [a] -> Maybe (NonEmpty a)`](https://www.stackage.org/haddock/lts-18.24/base-4.14.3.0/Data-List-NonEmpty.html#v:nonEmpty). (Given that this is "get _values_ from records" (plural), do you only want the first value?)

getValuesFromRecords :: [SearchResult] -> Maybe (NonEmpty Text)
getValuesFromRecords rrs = PDNS.sr_content <$> nonEmpty rrs

-- Lots of ways to phrase this one, so one example:
getOneValueFromRecords :: [SearchResult] -> Maybe (NonEmpty Text)
getOneValueFromRecords rrs = do
  (x :| _) <- nonEmpty rrs
  return $ PDNS.sr_content x